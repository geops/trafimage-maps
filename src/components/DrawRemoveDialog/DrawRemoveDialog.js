import React, { useCallback } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useTranslation } from 'react-i18next';
import { Button, makeStyles } from '@material-ui/core';
import PermalinkInput from '../PermalinkInput';
import Dialog from '../Dialog';
import { setDialogVisible, setDrawIds } from '../../model/app/actions';

export const NAME = 'drawRemoveDialog';

const useStyles = makeStyles((theme) => ({
  dialogContainer: {
    width: '400px !important',
  },
  link: {
    marginTop: theme.spacing(2),
  },
  footer: {
    marginTop: theme.spacing(2),
    paddingTop: theme.spacing(2),
    display: 'flex',
    justifyContent: 'flex-end',
    width: '100%',

    '& > button': {
      marginLeft: theme.spacing(2),
    },
  },
}));

function DrawRemoveDialog() {
  const classes = useStyles();
  const { t } = useTranslation();
  const dispatch = useDispatch();
  const drawLayer = useSelector((state) => state.map.drawLayer);

  const clear = useCallback(() => {
    drawLayer.olLayer.getSource().clear(true);
    dispatch(setDrawIds());
    dispatch(setDialogVisible());
  }, [drawLayer, dispatch]);

  const closeDialog = useCallback(() => {
    dispatch(setDialogVisible());
  }, [dispatch]);

  return (
    <Dialog
      isModal
      name={NAME}
      title={<span>{t('Zeichnung  löschen')}</span>}
      className={`tm-dialog-container ${classes.dialogContainer}`}
      body={
        <div>
          <div>
            {t('Wollen Sie den Plan wirklich aus der Karte löschen?')}
            <br />
            {t(
              'Um den Plan später zu bearbeiten, speichern Sie bitte folgenden Link:',
            )}
          </div>
          <div className={classes.link}>
            <PermalinkInput value={window.location.href} />
          </div>
          <div className={classes.footer}>
            <Button onClick={clear}>{t('Löschen')}</Button>
            <Button onClick={closeDialog} color="secondary">
              {t('Abbrechen')}
            </Button>
          </div>
        </div>
      }
    />
  );
}

export default React.memo(DrawRemoveDialog);
