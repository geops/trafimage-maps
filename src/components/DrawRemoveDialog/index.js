export { default, NAME } from './DrawRemoveDialog';
