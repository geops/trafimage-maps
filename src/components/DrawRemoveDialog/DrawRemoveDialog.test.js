import 'jest-canvas-mock';
import React from 'react';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import { Layer } from 'mobility-toolbox-js/ol';
import OLLayer from 'ol/layer/Vector';
import DrawRemoveDialog from '.';

describe('DrawRemoveDialog', () => {
  const mockStore = configureStore([thunk]);
  let store;

  beforeEach(() => {
    store = mockStore({
      app: {
        dialogPosition: {
          top: 0,
          left: 0,
        },
      },
      map: {
        drawlayer: new Layer({
          name: 'test',
          olLayer: new OLLayer(),
          properties: {
            description: 'description<br/>break',
          },
        }),
      },
    });
  });

  test('should match snapshot', () => {
    const component = renderer.create(
      <Provider store={store}>
        <DrawRemoveDialog />
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
