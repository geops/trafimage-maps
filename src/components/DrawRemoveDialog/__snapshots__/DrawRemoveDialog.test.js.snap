// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawRemoveDialog should match snapshot 1`] = `
<div>
  <div role="presentation"
       name="drawRemoveDialog"
       class="MuiDialog-root MuiModal-root css-4uaxrt-MuiModal-root-MuiDialog-root"
  >
    <div aria-hidden="true"
         class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
         style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
    </div>
    <div tabindex="0"
         data-testid="sentinelStart"
    >
    </div>
    <div class="MuiDialog-container MuiDialog-scrollPaper makeStyles-scrollPaperModal-9 css-1sep8xo-MuiDialog-container"
         style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
         role="presentation"
         tabindex="-1"
    >
      <div class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiDialog-paper makeStyles-paperModal-6 tm-dialog-container makeStyles-dialogContainer-1 MuiDialog-paperScrollPaper MuiDialog-paperWidthMd css-1hvry59-MuiPaper-root-MuiDialog-paper"
           role="dialog"
           aria-labelledby=":r0:"
      >
        <h2 class="MuiTypography-root MuiTypography-h4 MuiDialogTitle-root css-x5n03r-MuiTypography-root-MuiDialogTitle-root"
            id="draggable-dialog-title"
            style="cursor: auto;"
        >
          <span>
            Zeichnung löschen
          </span>
        </h2>
        <button class="MuiButtonBase-root Mui-focusVisible MuiIconButton-root MuiIconButton-sizeSmall makeStyles-closeBtn-13 css-1perq7w-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
                title="Dialog schließen"
        >
          <svg stroke="currentColor"
               fill="currentColor"
               stroke-width="0"
               viewbox="0 0 24 24"
               focusable="false"
               alt="Schliessen"
               height="1em"
               width="1em"
               xmlns="http://www.w3.org/2000/svg"
          >
            <path fill="none"
                  d="M0 0h24v24H0z"
            >
            </path>
            <path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z">
            </path>
          </svg>
        </button>
        <div class="makeStyles-dialogBody-10 ">
          <div>
            <div>
              Wollen Sie den Plan wirklich aus der Karte löschen?
              <br>
              Um den Plan später zu bearbeiten, speichern Sie bitte folgenden Link:
            </div>
            <div class="makeStyles-link-2">
              <div class="makeStyles-input-16">
                <input type="text"
                       tabindex="0"
                       title
                       class="tm-permalink-input"
                       value
                >
                <div role="button"
                     tabindex="0"
                     class="tm-permalink-bt"
                     title
                     aria-label
                >
                  <svg stroke="currentColor"
                       fill="currentColor"
                       stroke-width="0"
                       viewbox="0 0 24 24"
                       focusable="false"
                       height="1em"
                       width="1em"
                       xmlns="http://www.w3.org/2000/svg"
                  >
                    <path fill="none"
                          d="M0 0h24v24H0z"
                    >
                    </path>
                    <path d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z">
                    </path>
                  </svg>
                </div>
              </div>
            </div>
            <div class="makeStyles-footer-3">
              <button class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation css-1wggy8g-MuiButtonBase-root-MuiButton-root"
                      tabindex="0"
                      type="button"
              >
                Löschen
              </button>
              <button class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary MuiButton-disableElevation css-3vpgl4-MuiButtonBase-root-MuiButton-root"
                      tabindex="0"
                      type="button"
              >
                Abbrechen
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div tabindex="0"
         data-testid="sentinelEnd"
    >
    </div>
  </div>
</div>
`;
